client/extensions/woocommerce/state/sites/settings/products/actions.js

```diff
@@ -40,3 +43,39 @@ export const fetchSettingsProducts = siteId => ( dispatch, getState ) => {
 			dispatch( setError( siteId, getAction, err ) );
 		} );
 };
+
+/**
+ * Updates settings/products via batch endpoint.
+ *
+ * @param {Number} siteId wpcom site id.
+ * @param {Mixed}  settingsData, either single object { id: '', value: '' }, or array of settings objects
+ * @return {Object} Action object
+ */
+export const updateSettingsProducts = ( siteId, settingsData ) => dispatch => {
+	const updateData = Array.isArray( settingsData ) ? settingsData : [ settingsData ];
+	const updateAction = {
+		type: WOOCOMMERCE_SETTINGS_PRODUCTS_UPDATE_REQUEST,
+		data: settingsData,
```

Comment:
Did you want to pass `updateData` instead?

Reply:
yeah i went back and forth on that, but in the end decided to just keep the data passed in - alternatively I could require an array to be passed in instead.